engines/efh/fight.cpp

```diff
@@ -0,0 +1,1773 @@
+				}
+				// Action A - Add armor absorb text - End
+
+				if (noticedFl)
+					_messageToBePrinted += Common::String("  Your actions do not go un-noticed...");
+
+				// Action A - Check item durability - Start
+				int16 npcId = _teamCharId[teamCharId];
+
+				// get equipped inventory slot with exclusiveType == 9
+				uint16 exclusiveInventoryId = getEquippedExclusiveType(npcId, 9, false);
+				if (exclusiveInventoryId != 0x7FFF && _npcBuf[npcId]._inventory[exclusiveInventoryId].getUsesLeft() != 0x7F) {
+					int16 usesLeft = _npcBuf[npcId]._inventory[exclusiveInventoryId].getUsesLeft();
+					--usesLeft;
+					if (usesLeft <= 0) {
+						_messageToBePrinted += Common::String::format("  * %s%s's %s breaks!", _enemyNamePt1.c_str(), _enemyNamePt2.c_str(), _nameBuffer.c_str());
+						setCharacterObjectToBroken(npcId, exclusiveInventoryId);
+						var6E = false;
+					} else {
+						_npcBuf[npcId]._inventory[exclusiveInventoryId]._stat1 = (_npcBuf[npcId]._inventory[exclusiveInventoryId]._stat1 & 80) + usesLeft;
+					}
+				}
+				// Action A - Check item durability - End
+
+				// Action A - Check effect - Start
+				if (_items[teamCharItemId]._specialEffect == 1 && _mapMonsters[_techId][_teamMonsterIdArray[groupId]]._hitPoints[ctrMobsterId] > 0) {
+					if (getRandom(100) < 35) {
+						_teamMonsterEffects[groupId]._effect[ctrMobsterId] = 1;
+						_teamMonsterEffects[groupId]._duration[ctrMobsterId] = getRandom(10);
+						_messageToBePrinted += Common::String::format("  %s%s falls asleep!", _characterNamePt1.c_str(), _characterNamePt2.c_str());
+					}
+				} else if (_items[teamCharItemId]._specialEffect == 2 && _mapMonsters[_techId][_teamMonsterIdArray[groupId]]._hitPoints[ctrMobsterId] > 0) {
+					_teamMonsterEffects[groupId]._effect[ctrMobsterId] = 2;
+					_teamMonsterEffects[groupId]._duration[ctrMobsterId] = getRandom(10);
+					_messageToBePrinted += Common::String::format("  %s%s is frozen!", _characterNamePt1.c_str(), _characterNamePt2.c_str());
+				}
+				// Action A - Check effect - End
+			} else {
+				_messageToBePrinted = Common::String::format("%s%s tries to use %s %s, but it doesn't work!", _enemyNamePt1.c_str(), _enemyNamePt2.c_str(), kPossessive[ennemyPronoun], _nameBuffer.c_str());
+			}
+
+			genericGenerateSound(_items[teamCharItemId]._attackType, hitCount);
+			displayBoxWithText(_messageToBePrinted, 1, 2, true);
+		}
+	}
+}
+
+void EfhEngine::handleFight_lastAction_D(int16 teamCharId) {
+	// Fight - Action 'D' - Defend
+	// In the original, this function is part of handleFight.
+	// It has been split for readability purposes.
+	debugC(3, kDebugFight, "handleFight_lastAction_D %d", teamCharId);
+
+	_teamPctDodgeMiss[teamCharId] -= 40;
+	_enemyNamePt2 = _npcBuf[_teamCharId[teamCharId]]._name;
+
+	uint8 pronoun = _npcBuf[_teamCharId[teamCharId]].getPronoun();
+
+	if (pronoun == 2)
```

Comment:
This cries for `_enemyNamePt1 = getArticle(pronoun);` and in other places.

Reply:
no, because it's used later in the message to be printed : kPersonal[pronoun]